pkgs/development/libraries/itk/generic.nix

```diff
@@ -63,20 +71,31 @@ stdenv.mkDerivation {
     "-DModule_GenericLabelInterpolator=ON"
   ];
 
-  nativeBuildInputs = [ cmake xz makeWrapper ];
-  buildInputs = [ libX11 libuuid vtk ] ++ lib.optionals stdenv.isDarwin [ Cocoa ];
+  nativeBuildInputs = [ cmake xz ];
+  buildInputs = [
+    libX11
+    libuuid
+    vtk
+  ] ++ lib.optionals stdenv.isDarwin [ Cocoa ];
   # Due to ITKVtkGlue=ON and the additional dependencies needed to configure VTK 9
   # (specifically libGL and libX11 on Linux),
   # it's now seemingly necessary for packages that configure ITK to
   # also include configuration deps of VTK, even if VTK is not required or available.
   # These deps were propagated from VTK 9 in https://github.com/NixOS/nixpkgs/pull/206935,
   # so we simply propagate them again from ITK.
   # This admittedly is a hack and seems like an issue with VTK 9's CMake configuration.
-  propagatedBuildInputs = vtk.propagatedBuildInputs;
-
-  postInstall = ''
-    wrapProgram "$out/bin/h5c++" --prefix PATH ":" "${pkg-config}/bin"
-  '';
+  propagatedBuildInputs = [
+    # Unvendored dependencies such as GDCM must be propagated:
```

Comment:
Why is that?

Reply:
If not, libraries build against ITK fail at configure time (unless all these ITK deps are added as buildInputs, which is anti-modular) since ITK still expects to be able to find the relevant headers (and presumably the same issue would come up at build/link time as well).

Reply:
Pushed an update to this comment explaining this.

Reply:
Normally I'd expect pkg-config taking care of that, but I guess there's none here then

Reply:
No, pkg-config doesn't seem to be sufficient here.